### PR TITLE
fix(1701): PR authors should not be able to restart builds if `restrictPR` is on

### PR DESCRIPTION
**Pull request by klu909**
## Context

PR author should not be allowed to restart PR if restrictPR annotation is turned on.

## Objective

Add a check to enforce restrictPR before allowing PR author to restart builds

## References

https://github.com/screwdriver-cd/screwdriver/issues/1701

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/events/create.js [hunk 1]
```diff
@@ -159,9 +162,11 @@ module.exports = () => ({
                                         payload.prInfo = prInfo;
                                         payload.prRef = prInfo.ref;
                                         payload.chainPR = pipeline.chainPR;
+                                        const restrictPR = pipeline.annotations[ANNOT_RESTRICT_PR];
 
-                                        // PR author should be able to rerun their own PR build
-                                        if (prInfo.username === username) {
+                                        // PR author should be able to rerun their own PR build if restrictPR is not on
```
Comment (jithin1987) [inline plugins/events/create.js]: there is a value `fork` also for restrictPR

File: plugins/events/create.js [hunk 2]
```diff
@@ -4,6 +4,9 @@ const boom = require('boom');
 const urlLib = require('url');
 const validationSchema = require('screwdriver-data-schema');
 
+const ANNOT_NS = 'screwdriver.cd';
+const ANNOT_RESTRICT_PR = `${ANNOT_NS}/restrictPR`;
```
Comment (jithin1987) [inline plugins/events/create.js]: we don't need ANNOT_NS to be a separate variable

File: plugins/events/create.js (hunk 1 above)
Comment (klu909) [reply plugins/events/create.js]: can you elaborate on this?

Comment (tkyi) [reply plugins/events/create.js]: https://docs.screwdriver.cd/user-guide/configuration/annotations#pipeline-level-annotations

**Pull request merged**